test/SlowTests/Server/Documents/DataArchival/DataArchivalTests.cs

```diff
@@ -258,6 +259,199 @@ public async Task ArchiveFlagShouldNotBeRemoved()
                 }
             }
         }
+        
+        [RavenTheory(RavenTestCategory.ExpirationRefresh)]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task ShouldNotApplyArchivedMetadataForNewDocuments(bool archivedValue)      
+        {
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    var metadataPropNameToTest = "@archived";
+                    var executor = store.GetRequestExecutor();
+                    using var dis = executor.ContextPool.AllocateOperationContext(out var context);
+                    var p = context.ReadObject(new DynamicJsonValue
```

Comment:
why are you putting docs like that? cant we do `session.GetMetadataFor(entity)` ?